### PR TITLE
gadget: mv ensureLayoutCompatibility to gadget proper, add gadgettest pkg

### PR DESCRIPTION
**Pull request by anonymouse64**
Move ensureLayoutCompatibility and related helpers to the top-level gadget
package so that they can be used from the gadget asset update routine.

Also move some gadget test helpers that are now shared between multiple gadget
test packages (install_test and gadget_test now) to a shared gadgettest package
where we can easily share gadget helpers. This package can probably grow more
helpers but this is the bare minimum needed to move forward.

Note that none of the logic has changed anywhere, this is a pure mechanical move,
though some of the helpers in gadgettest package are changed slightly since that doesn't
import go-check.

Hopefully the introduction of the gadgettest package is non-controversial, I didn't realize 
I needed to do this when I discussed moving ensureLayoutCompatibility around with
@pedronis, so the review here is just to make sure that this is "okay". I don't think a 
full review here is necessary from pedronis, just +1 on creation of the gadgettest package
really.

### CODE REVIEW

File: gadget/gadgettest/gadgettest.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/gadget"
+	"github.com/snapcore/snapd/gadget/quantity"
+)
+
+func LayoutFromYaml(newDir, gadgetYaml string, model gadget.Model) (*gadget.LaidOutVolume, error) {
```
Comment (pedronis) [inline gadget/gadgettest/gadgettest.go:33]: this probably merits a doc comment also about the expectation that gadgetYaml must have one volume?

Comment (anonymouse64) [reply gadget/gadgettest/gadgettest.go:33]: yes good point it should have a similar doc-comment to the other function

**Review by stolowski — APPROVED** [review]
Looks fine and sensible, thank you

**Review by pedronis — APPROVED** [review]
lgtm

**Pull request merged**